sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-text.models.ts

```diff
@@ -0,0 +1,124 @@
+import { Contains, IsDate, IsNumberString, Length } from "class-validator";
+import { APPLICATION_NUMBER_LENGTH } from "@sims/sims-db";
+import { IsValidSIN } from "../../utilities/class-validation";
+import { getDateOnlyFromFormat } from "@sims/utilities";
+
+export const DATE_FORMAT = "YYYYMMDD";
+/**
+ * user-friendly header names used in the text to be populated by the user.
+ * The text model parser uses this as a base to parse the text string into an object model.
+ */
+export const DataTextHeaders = {
+  recordType: "Record Type",
+  sin: "SIN",
+  applicationNumber: "Application Number",
+  withdrawalDate: "Withdrawal Date",
+};
+
+export enum RecordType {
+  ApplicationBulkWithdrawalHeaderRecordType = "1",
+  ApplicationBulkWithdrawalDataRecordType = "2",
+  ApplicationBulkWithdrawalFooterRecordType = "3",
+}
+
+export class ApplicationWithdrawalTextModel {
+  /**
+   * Data record type.
+   */
+  @Contains(RecordType.ApplicationBulkWithdrawalDataRecordType, {
+    message: `${DataTextHeaders.recordType} must be valid.`,
+  })
+  recordType: RecordType;
+  /**
+   * SIN.
+   */
+  @IsValidSIN({
```

Comment:
I believe we spoke about not doing valid sin check because SIN is going to be used later in the query. As per the conversion we must check for valid number and length of 9 characters.